Tools/cases_generator/analyzer.py

```diff
@@ -1079,6 +1081,36 @@ def add_instruction(name: str) -> None:
     return instmap, len(no_arg), min_instrumented
 
 
+def is_instruction_size_macro(token: lexer.Token) -> bool:
+    return token.kind == "IDENTIFIER" and token.text == "INSTRUCTION_SIZE"
+
+
+def get_instruction_size_for_uop(instructions: dict[str, Instruction], uop: Uop) -> int | None:
+    """Return the size of the instruction that contains the given uop or
+    `None` if the uop does not contains the `INSTRUCTION_SIZE` macro.
+
+    If there is more than one instruction that contains the uop,
+    ensure that they all have the same size.
+    """
+    if not (token := next((t for t in uop.body if is_instruction_size_macro(t)), None)):
```

Comment:
```suggestion
    for tkn in uop.body:
          if tkn.text == "INSTRUCTION_SIZE":
               break
     else:
```

Using a generator expression, `next`, and the walrus operator makes this a bit hard to follow. Maybe use a plain for loop?